lib/cosmos/interfaces/interface.rb

```diff
@@ -372,7 +372,7 @@ def convert_data_to_packet(data)
     # @param packet [Packet] Packet to extract data from
     # @return data
     def convert_packet_to_data(packet)
-      packet.buffer(false)
+      packet.buffer(true) # Copy buffer so logged command isn't modified
```

Comment:
basically write_packet does modify the packet that gets logged.  But write_data shouldn't because it could do things like expand the size of the buffer which is bad and we don't want logged